app/src/main/java/com/keylesspalace/tusky/adapter/EmojiFontAdapter.java

```diff
@@ -0,0 +1,317 @@
+import java.io.InputStreamReader;
+import java.lang.reflect.Type;
+import java.util.ArrayList;
+
+import static android.view.View.GONE;
+
+/**
+ * This adapter is used to provide the items in the emoji style picker
+ */
+public class EmojiFontAdapter extends RecyclerView.Adapter<EmojiFontAdapter.EmojiFontViewHolder>
+        implements EmojiPreference.EmojiFontListener {
+
+    // All fonts available in this list
+    private ArrayList<EmojiCompatFont> fonts = new ArrayList<>();
+    // The currently selected font which will be chosen when clicking OK
+    private EmojiCompatFont selected;
+    // All Radio buttons.
+    // Since a RadioGroup can't be used, unchecking all the other radio buttons needs to be done manually.
+    private final ArrayList<RadioButton> radioButtons = new ArrayList<>();
+    // The directory containing all the emoji files
+    private final File dlDir;
+
+    /**
+     * Create a new EmojiFontAdapter to fill the selector GUI
+     * @param dlDir The directory with all the emoji files
+     * @param selected the currently selected emoji font (maybe system default?)
+     */
+    public EmojiFontAdapter(File dlDir, EmojiCompatFont selected) {
+        super();
+        this.dlDir = dlDir;
+        this.selected = selected;
+    }
+
+    /**
+     * Load a list of fonts from a JSON file
+     * @param fontList the file containing the JSON document
+     * @return An ArrayList containing the EmojiCompatFonts represented in the JSON file
+     * @throws FileNotFoundException Thrown if the JSON file can't be found
+     */
+    private ArrayList<EmojiCompatFont> parseFonts(File fontList) throws FileNotFoundException {
+        // Use Gson
+        Gson gson = new Gson();
+        // This is actually copied from their user guide
+        Type emojiFontType = new TypeToken<ArrayList<EmojiCompatFont>>(){}.getType();
+        ArrayList<EmojiCompatFont> fonts;
+        try {
+            fonts =
+                    gson.fromJson(new InputStreamReader(new FileInputStream(fontList)), emojiFontType);
+        }
+        catch (JsonSyntaxException ex) {
+            ex.printStackTrace();
+            return new ArrayList<>();
+        }
+        // The font objects don't have their base directory yet...
+        for(EmojiCompatFont font: fonts) {
+            font.setBaseDirectory(dlDir);
+        }
+        // That's it!
+        return fonts;
+    }
+
+    /**
+     * Create a new skeleton for an entry
+     * @param parent Probably the RecyclerView using this adapter
+     * @param viewType ????
+     * @return An empty font entry
+     */
+    @NonNull
+    @Override
+    public EmojiFontViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
+        View item = LayoutInflater.from(parent.getContext())
+                .inflate(R.layout.emoji_pref_item, parent, false);
+        // All radio buttons need to be stored to uncheck them later
+        radioButtons.add(item.findViewById(R.id.emoji_radio));
```

Comment:
I think this does not work with views that are recycled

Reply:
As far as I remember onCreateViewholder is never called on recycled Views as this creates these Views.  
There's another method for recycled Views which I don't remember...